### PR TITLE
Set confirmation page column coordinate to after the CYA page

### PR DESCRIPTION
**Pull request by njseeto**
[Trello](https://trello.com/c/c4I4IvM3/2410-sometime-the-flow-view-shows-confirmation-before-cya-and-other-bugs)
- If confirmation and CYA pages exist, deliberately set the confirmation page to be after the CYA page.
- Bump to 2.16.2

### Before
before

### After
after

### CODE REVIEW

File: app/models/metadata_presenter/grid.rb [hunk 1]
```diff
@@ -225,6 +231,12 @@ def detached?(position)
       position[:row].nil? || position[:column].nil?
     end
 
+    def confirmation_and_cya_pages_present?
+      service.confirmation_page.present? &&
+        service.checkanswers_page.present? &&
+        !coordinates.positions[service.checkanswers_page.uuid][:column].nil?
```
Comment (brenetic) [inline app/models/metadata_presenter/grid.rb:237]:
```suggestion
        coordinates.positions[service.checkanswers_page.uuid][:column].present?
```

File: app/models/metadata_presenter/grid.rb [hunk 2]
```diff
@@ -215,6 +215,12 @@ def add_by_coordinates
         next if detached?(position)
 
         column = position[:column]
+
+        # Add confirmation page column co-ordinate
```
Comment (brenetic) [inline app/models/metadata_presenter/grid.rb:219]:
```suggestion
        # Update confirmation page column co-ordinate
```

Comment (brenetic) [reply app/models/metadata_presenter/grid.rb:219]: think maybe an if else might be good

File: app/models/metadata_presenter/grid.rb (hunk 1 above)
Comment (brenetic) [reply app/models/metadata_presenter/grid.rb:237]: if only the confirmation page is present i assume the grid just places the confirmation page wherever the column model calculated it?

**Comment by brenetic** [comment]
i think we can have some tests around this, maybe in the grid _spec

File: app/models/metadata_presenter/grid.rb (hunk 1 above)
Comment (njseeto) [reply app/models/metadata_presenter/grid.rb:237]: yep correct, the grid will place the confirmation page where the column model has calculated.

**Review by brenetic — APPROVED** [review]
🚀

**Pull request merged**